### PR TITLE
Fix:Goroutine leaks when getting audit event sender times out

### LINKED ISSUE #5341: Goroutine leaks when getting audit event sender times out
**Describe the Bug**
Goroutine leaks when getting audit event sender times out.

https://github.com/kubesphere/kubesphere/blob/7162d4131042e318c178690c5856ce64ff158ae5/pkg/apiserver/auditing/backend.go#L149-L154
https://github.com/kubesphere/kubesphere/blob/7162d4131042e318c178690c5856ce64ff158ae5/pkg/apiserver/auditing/backend.go#L184-L186

**Versions Used**
KubeSphere:v3.3.1

**Environment**

**How To Reproduce**
When the maximum event senders number is reached

**Expected behavior**
No goroutines leak

### PR DESCRIPTION
**Pull request by hzhhong**
### What type of PR is this?

/kind bug

### What this PR does / why we need it:
Goroutine leaks when getting audit event sender times out.

### Which issue(s) this PR fixes:

Fixes #5341

### Special notes for reviewers:
```
```

### Does this PR introduced a user-facing change?

```release-note
None
```

### Additional documentation, usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/apiserver/auditing/backend.go [hunk 1]
```diff
@@ -183,6 +185,9 @@ func (b *Backend) sendEvents(events *v1alpha1.EventList) {
 	go send()
 
 	defer func() {
+		if skipReturnSender {
```
Comment (wanjunlei) [inline pkg/apiserver/auditing/backend.go:188]: I think this is more readable.

```
		if !skipReturnSender {
			<-b.senderCh
		}
```

**Comment by hzhhong** [comment]
PTAL @wanjunlei

**Review by wanjunlei — APPROVED** [review]

**Comment by hzhhong** [comment]
PTAL @wansir

**Pull request merged**